service/controller/resource/clusterrolebinding/update.go

```diff
@@ -21,7 +23,43 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 		// Create the cluster role bindings in the Kubernetes API.
 		for _, clusterRoleBinding := range clusterRoleBindingsToUpdate {
 			_, err := r.k8sClient.RbacV1().ClusterRoleBindings().Update(clusterRoleBinding)
-			if err != nil {
+			if isExternalFieldImmutableError(err) {
+				// We can't change a RoleRef, so create a temporary CRB which references the new RoleRef
+				newCRB := apiv1.ClusterRoleBinding{}
+
+				clusterRoleBinding.DeepCopyInto(&newCRB)
+				temporaryName := clusterRoleBinding.Name + "-upgrading"
+				newCRB.SetName(temporaryName)
+
+				_, err := r.k8sClient.RbacV1().ClusterRoleBindings().Create(&newCRB)
+				if apierrors.IsAlreadyExists(err) {
+				} else if err != nil {
+					return microerror.Mask(err)
+				}
+				// Try deleting and then move to re-create-crb branch
+				// Delete the old CRB
+				err = r.k8sClient.RbacV1().ClusterRoleBindings().Delete(clusterRoleBinding.Name, &apismetav1.DeleteOptions{})
+				if apierrors.IsNotFound(err) {
+				} else if err != nil {
+					return microerror.Mask(err)
+				}
+
+				// Create a CRB with the original name, but use the new configuration
+				newCRB.SetName(clusterRoleBinding.Name)
```

Comment:
We talked and this name should be the one coming from the `helm` template.

It seems like from testing that the `-upgrading` one is not needed at all. If I understood @stone-z correctly.